src/azure-cli/azure/cli/command_modules/appservice/static_sites.py

```diff
@@ -192,9 +193,13 @@ def update_staticsite_users(cmd, name, roles, authentication_provider=None, user
 def create_staticsites(cmd, resource_group_name, name, location,
                        source, branch, token=None,
                        app_location='.', api_location='.', output_location='.github/workflows',
-                       tags=None, no_wait=False, sku='Free'):
-    if not token:
+                       tags=None, no_wait=False, sku='Free', login_with_github=False):
+    if not token and not login_with_github:
         _raise_missing_token_suggestion()
+    elif not token:
+        from ._github_oauth import get_github_access_token
+        scopes = ["admin:repo_hook", "repo", "workflow"]
+        token = get_github_access_token(cmd, scopes)
```

Comment:
What if `token and login_with_github`? `not token and not login_with_github`?

Reply:
Then `token` will be used and login_with_github param will be ignored

Reply:
Maybe we can give a warning or add it to parameter help.